code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -447,6 +445,8 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 			. |= reaction.react(src, dump_location)
 	if(.)
 		garbage_collect()
+		if(temperature < TCMB) //just for safety
+			temperature = TCMB

```

Comment:
these checks being moved is a minor performance boost, maintains the same level of sanity, and makes the primary check of this PR functional